morphir/src/org/finos/morphir/datamodel/UnionType.scala

```diff
@@ -0,0 +1,8 @@
+package org.finos.morphir.datamodel
+
+sealed trait UnionType
```

Comment:
What is the difference between `SealedTrait`, `Enum`, and `Sum`?

Reply:
When I used reflection to construct them I needed to be able to tell Scala 3 Enum and SealedTrait apart. Now since I use Mirror.ordinal it's not needed. I'll keep in mind to remove one of the cases.